pages/api/items/[id].js

```diff
@@ -5,33 +5,36 @@ export default async function handler(request, response) {
   await dbConnect();
 
   const { id } = request.query;
-
-  if (request.method === "GET") {
-    try {
+  try {
+    if (request.method === "GET") {
       const item = await Item.findById(id);
       if (!item) {
         return response.status(404).json({ message: "Item not found" });
       }
       return response.status(200).json(item);
-    } catch (error) {
-      return response.status(500).json({ message: "Server error" });
     }
-  }
 
-  if (request.method === "DELETE") {
-    try {
+    if (request.method === "DELETE") {
       const deleteShoppingItem = await Item.findByIdAndDelete(id);
       if (!deleteShoppingItem) {
         return response.status(404).json({ message: "Item not found" });
       }
       return response
         .status(200)
         .json({ message: "Successfully deleted this item" });
-    } catch (error) {
-      console.error(error);
-      return response.status(500).json({ message: "Internal Server Error." });
     }
+
+    if (request.method === "PUT") {
+      const updatedItem = await Item.findByIdAndUpdate(id, request.body);
+      if (!updatedItem) {
+        return response.status(404).json({ message: "Item not found" });
```

Comment:
are you aware of the fact that you don't need to return the response object from the function in order to send it? The `return` keyword is only used to prevent further code execution inside the handler function. Like in an early return statement. By writing your code like this it is implied that you want the response to be returned from the function, which does not make sense. This will confuse new devs when they read the code. it would be more appropriate to split these into two separate lines to highlight, that we in fact don't return anything from the functino but rather stopping the execution after sending the response:
```suggestion
        response.status(404).json({ message: "Item not found" });
        return; 
```